src/client.rs

```diff
@@ -399,3 +414,100 @@ pub enum ListOffsetsPosition {
     /// Fetch the offset for the corresponding timestamp.
     Timestamp(i64),
 }
+
+pub struct Admin {
+    /// The client handle from which this producer was created.
+    _client: Client,
+    /// The channel used by this producer.
+    tx: mpsc::UnboundedSender<BrokerResponse>,
+    /// The channel used by this producer.
+    rx: mpsc::UnboundedReceiver<BrokerResponse>,
+    /// Discovered metadata on a Kafka cluster along with broker connections.
+    ///
+    /// NOTE WELL: this value should never be updated outside of the `ClientTask`.
+    cluster: ClusterMeta,
```

Comment:
I've added the `get_cluster_metadata_cache()` method on the underlying client which handles all of this, and will also keep it up-to-date. 

Maybe we should just cut over to using that instead of maintaining a separate symbol there. Thoughts?